src/convnets/shufflenet.jl

```diff
@@ -0,0 +1,160 @@
+using Flux
+
+"""
+Channelshuffle(channels, groups)
+
+Channel shuffle operation from 'ShuffleNet: An Extremely Efficient Convolutional Neural Network for Mobile Devices
+([reference](https://arxiv.org/abs/1707.01083)).
+
+# Arguments
+
+  - `channels`: number of channels
+  - `groups`: number of groups
+"""
+function ChannelShuffle(x::Array{Float32, 4}, g::Int)
+    width, height, channels, batch = size(x)
+    channels_per_group = channels÷g
+    if (channels % g) == 0
+        x = reshape(x, (width, height, g, channels_per_group, batch))
+        x = permutedims(x,(1,2,4,3,5))
+        x = reshape(x, (width, height, channels, batch))
+    end
+    return x
+end
+
+"""
+ShuffleUnit(in_channels::Integer, out_channels::Integer, grps::Integer, downsample::Bool, ignore_group::Bool)
+
+Shuffle Unit from 'ShuffleNet: An Extremely Efficient Convolutional Neural Network for Mobile Devices
+([reference](https://arxiv.org/abs/1707.01083)).
+
+# Arguments
+
+  - `in_channels`: number of input channels
+  - `out_channels`: number of output channels
+  - `groups`: number of groups
+  - `downsample`: apply downsaple if true
+  - `ignore_group`: ignore group convolution if true
+"""
+function ShuffleUnit(in_channels::Integer, out_channels::Integer, groups::Integer, downsample::Bool, ignore_group::Bool)
+    mid_channels = out_channels ÷ 4
+    groups = ignore_group ? 1 : groups
+    strd = downsample ? 2 : 1
+
+    if downsample
+        out_channels -= in_channels
+    end
+
+    m = Chain(Conv((1,1), in_channels => mid_channels; groups,pad=SamePad()),
+              BatchNorm(mid_channels),
+              NNlib.relu,
```

Comment:
```suggestion
              BatchNorm(mid_channels, relu),
```
`relu` is already in scope because of `using NNlib` and fusing it into the preceeding norm is slightly more efficient. Also, is the activation function not configurable for ShuffleNet?

Reply:
Changed.